externs.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @constructor
+ * @param {(string|!NodeName)} nodeName
+ * @param {Object|undefined} attributes
```

Comment:
`@param {Object=} opt_attributes`

(= is "this or undefined". Also currently this is `null`, `Object` or `undefined`. You can say `{!Object=}` to be `Object` or `undefined`.